homeassistant/components/sensor/fail2ban.py

```diff
@@ -61,13 +58,17 @@ def __init__(self, name, jail, log_parser):
         """Initialize the sensor."""
         self._name = '{} {}'.format(name, jail)
         self.jail = jail
-        self.ban_dict = {STATE_CURRENT_BANS: [], STATE_ALL_BANS: []}
-        self.last_ban = None
+        self.ban_dict = {STATE_LAST_BAN: None,
+                         STATE_CURRENT_BANS: [], STATE_ALL_BANS: []}
+        self.number = 0
         self.log_parser = log_parser
         self.log_parser.ip_regex[self.jail] = re.compile(
-            r"\[{}\].(Ban|Unban) ([\w+\.]{{3,}})".format(re.escape(self.jail))
+            r"\[{}\]\s*(Ban|Unban) (.*)".format(re.escape(self.jail))
         )
-        _LOGGER.debug("Setting up jail %s", self.jail)
+        _LOGGER.info("Setting up jail %s", self.jail)
+        _LOGGER.debug("Regex: %s",
+                      str(self.log_parser.ip_regex[self.jail]))
+        self.log_parser.read_log(self.jail)
```

Comment:
Update is called when adding the entity so reading the log here is not needed.

Reply:
I needed that with the previous time logic. I will remove that.